### DISCUSSION TITLE
CGAL::angle() applications

### DISCUSSION THREAD

**Discussion by petrasvestartas** [CGAL/cgal#5988] [Q&A] [answered]
Hi,

What do you usually use `CGAL:angle `method for?

I tried to search in CGAL repo but I cannot find explanation about:
` CGAL::OBTUSE, CGAL::RIGHT or CGAL::ACUT`
I guess that CGAL::Right is 90 degrees and CGAL::OBTUSE is x>90 degrees and CGAL::ACUT x<90.

Is it correct? And what do you usually use this for? Detection of concave or convex corners?

**Comment by lrineau** [comment, marked as answer]
## Usage in CGAL

`CGAL::angle` is not directly used, in CGAL. In the CGAL algorithm, we use the kernel API: `CGAL::Angle_2` and `CGAL::Angle_3`.
  - `Angle_2` is used in:
    - [`Mesh_2/include/CGAL/Mesh_2/Refine_edges.h`](https://github.com/CGAL/cgal/blob/8c9fc331cd0e523f897f830da480483b6fb70d5d/Mesh_2/include/CGAL/Mesh_2/Refine_edges.h#L158), and
    - [`Mesh_2/include/CGAL/Mesh_2/Clusters.h`](https://github.com/CGAL/cgal/blob/master/Mesh_2/include/CGAL/Mesh_2/Clusters.h#L538).
  - `Angle_3` is used in:
    - [`Mesh_3/include/CGAL/Mesh_domain_with_polyline_features_3.h`](https://github.com/CGAL/cgal/blob/8c9fc331cd0e523f897f830da480483b6fb70d5d/Mesh_3/include/CGAL/Mesh_domain_with_polyline_features_3.h#L428), and
    - [`Surface_mesh_segmentation/include/CGAL/Surface_mesh_segmentation/internal/SDF_calculation.h`](https://github.com/CGAL/cgal/blob/20ea9e775f33617a8ffc06c6be85c1b0acc3ab3e/Surface_mesh_segmentation/include/CGAL/Surface_mesh_segmentation/internal/SDF_calculation.h#L463).

## Meaning

> I tried to search in CGAL repo but I cannot find explanation about:
> `CGAL::OBTUSE`, `CGAL::RIGHT` or `CGAL::ACUTE`
> I guess that `CGAL::RIGHT` is 90 degrees and `CGAL::OBTUSE` is x>90 degrees and `CGAL::ACUTE` x<90.

Yes, that is right. That has the same meaning as the mathematical meaning: 
  - a right angle is 90°,
  - an acute angle is <90°,
  - and an obtuse angle is >90°.
